Clarify highlighting docs for semantic_text and interacting with chunks (#131871)
elastic/elasticsearch: docs/reference/elasticsearch/mapping-reference/semantic-text.md

```

1. Specifies the maximum number of fragments to return.

> 2. Sorts returned most relevant highlighted fragments by score when set to `score`. By default,

Reviewer: ```suggestion
2. Sorts the most relevant highlighted fragments by score when set to `score`. By default,
```

This is clearer to me because, if I understand correctly, "returned" refers to the fragments returned by the underlying highlighter context. However, this is a hidden implementation detail from the user's perspective.

   fragments will be output in the order they appear in the field (order: none).

> To use the `semantic` highlighter to view indexed chunks in the order which they
> were indexed and no scoring,
> use the `match_all` query to retrieve each chunk in the order which it appears
> in the document:

Reviewer: ```suggestion
To use the `semantic` highlighter to view chunks in the order which they were indexed with no scoring,
use the `match_all` query to retrieve them in the order they appear in the document:
```

Less word repetition with this phrasing

```console
POST test-index/_search
{
    "query": {

>         "match_all": {}

Author: @Samiul-TheSoccerFan what version should this be tagged with?

    },
    "highlight": {
        "fields": {
            "my_semantic_field": {

>                 "number_of_fragments": 5

Reviewer: Should we call out adjusting `number_of_fragments` here to a larger number to get all chunks in a doc, and how that will be variable?

Author: Updated in 8cae0f659b32d45327713e8c99ca0cea1783e95a LMKWYT
